application/classes/Ushahidi/Core.php

```diff
@@ -469,21 +469,23 @@ public static function init()
 		$di->set('repository.post.text', $di->lazyNew('Ushahidi_Repository_Post_Text'));
 		$di->set('repository.post.varchar', $di->lazyNew('Ushahidi_Repository_Post_Varchar'));
 		$di->set('repository.post.media', $di->lazyNew('Ushahidi_Repository_Post_Media'));
+		$di->set('repository.post.video', $di->lazyNew('Ushahidi_Repository_Post_Video'));
 
 		// The post value repo factory
 		$di->set('repository.post_value_factory', $di->lazyNew('Ushahidi_Repository_Post_ValueFactory'));
 		$di->params['Ushahidi_Repository_Post_ValueFactory'] = [
 				// a map of attribute types to repositories
 				'map' => [
-					'datetime' => $di->lazyGet('repository.post.datetime'),
-					'decimal'  => $di->lazyGet('repository.post.decimal'),
-					'geometry' => $di->lazyGet('repository.post.geometry'),
-					'int'      => $di->lazyGet('repository.post.int'),
-					'point'    => $di->lazyGet('repository.post.point'),
-					'relation' => $di->lazyGet('repository.post.relation'),
-					'text'     => $di->lazyGet('repository.post.text'),
-					'varchar'  => $di->lazyGet('repository.post.varchar'),
-					'media'    => $di->lazyGet('repository.post.media'),
+					'datetime'   => $di->lazyGet('repository.post.datetime'),
+					'decimal'    => $di->lazyGet('repository.post.decimal'),
+					'geometry'   => $di->lazyGet('repository.post.geometry'),
+					'int'        => $di->lazyGet('repository.post.int'),
+					'point'      => $di->lazyGet('repository.post.point'),
+					'relation'   => $di->lazyGet('repository.post.relation'),
+					'text'       => $di->lazyGet('repository.post.text'),
+					'varchar'    => $di->lazyGet('repository.post.varchar'),
+					'media'      => $di->lazyGet('repository.post.media'),
+					'video'      => $di->lazyGet('repository.post.video'),
```

Comment:
I think in theory you could just map this to either post.varchar or post.text .. and share the same table. We only really need custom validation right?